bridges/centralized-ethereum/src/actors/watch_dog.rs

```diff
@@ -0,0 +1,364 @@
+        wit_balance: Option<f64>,
+        ctx: &mut Context<Self>,
+        period: Duration,
+    ) {
+        if self.start_eth_balance.is_none() && eth_balance.is_some() {
+            self.start_eth_balance = eth_balance;
+        }
+        if let Some(wit_balance) = wit_balance {
+            if wit_balance > self.start_wit_balance.unwrap_or_default() {
+                self.start_wit_balance = Some(wit_balance);
+                log::warn!("Wit account refunded to {} $WIT", wit_balance);
+            }
+        }
+        let start_eth_balance = self.start_eth_balance;
+        let start_wit_balance = self.start_wit_balance;
+        let wit_jsonrpc_socket = self.wit_jsonrpc_socket.clone();
+        let wit_utxo_min_value_threshold = self.wit_utxo_min_value_threshold;
+        let eth_jsonrpc_url = self.eth_jsonrpc_url.clone();
+        let eth_account = self.eth_account;
+        let eth_contract_address = self.eth_contract.clone().unwrap().address();
+        let running_secs = self.start_ts.unwrap().elapsed().as_secs();
+
+        let fut = async move {
+            let mut status = "up-and-running".to_string();
+
+            if let Err(err) = check_wit_connection_status(&wit_jsonrpc_socket).await {
+                status = err;
+            }
+            let wit_client = JsonRpcClient::start(&wit_jsonrpc_socket)
+                .expect("cannot start JSON/WIT connection");
+            let wit_account = match fetch_wit_account(&wit_client).await {
+                Ok(pkh) => pkh,
+                Err(err) => {
+                    if status.eq("up-and-running") {
+                        status = err;
+                    }
+                    None
+                }
+            };
+
+            let wit_balance = match wit_account.clone() {
+                Some(pkh) => match fetch_wit_account_balance(&wit_client, pkh.as_str()).await {
+                    Ok(wit_balance) => wit_balance,
+                    Err(err) => {
+                        if status.eq("up-and-running") {
+                            status = err;
+                        }
+                        None
+                    }
+                },
+                None => None,
+            };
+
+            let wit_utxos_above_threshold = match wit_account.clone() {
+                Some(pkh) => {
+                    match fetch_wit_account_count_utxos_above(
+                        &wit_client,
+                        pkh.as_str(),
+                        wit_utxo_min_value_threshold,
+                    )
+                    .await
+                    {
+                        Ok(wit_utxos_above_threshold) => wit_utxos_above_threshold,
+                        Err(err) => {
+                            if status.eq("up-and-running") {
+                                status = err;
+                            }
+                            None
+                        }
+                    }
+                }
+                None => None,
```

Comment:
We could move this three calls into a `fetch_wit_info` function that returns `(wit_account, wit_balance, wit_utxos_above_threshold)`